### DISCUSSION TITLE
Changes are not reflecting in sunbird-ED portal

### DISCUSSION THREAD

**Discussion by israr-trigyn** [Sunbird-Ed/Community#622] [Bugs] [answered]
Hi There

We're facing issues related to Live-Reload in Sunbird-Ed.

Whenever we make any changes it does not reflect

Earlier when we were making any changes auto build generation was failing due to "momen.js" version issue, so we were told to downgrade the momen.js version from "2.29" to 2.24, and now after downgrading whenever we make changes build generates successfully, but still whatever we change that doesn't appear. but build generates successfully after downgrading moment.js to 2.24.

So as a result we're still on the same page earlier auto build was not generating and now build is generating but still changes not reflecting, We have to run build command again and again to see the changes.

**Comment by israr-trigyn** [comment]
@vinukumar-vs @rajeevsathish @anandp504 

Please response to above query

**Reply by rajeshkumaravel** [reply]
@israr-trigyn 
To work with hot reload
Please ensure the following steps
- Downgrade the moment version to 2.24.0 in package json of client (Both in dependencies and resolutions)
- Remove yarn lock file
- Delete `node_modules` folder (For Client)
- Run `yarn install`
- Then start the application, hot reload should work

_**Ensure not to checkin / contribute the package json with downgraded moment version**_

Watch for any error in client terminal, if so please share it here

CC: @vinukumar-vs @anandp504 @rajeevsathish @Krishnaj20

**Reply by israr-trigyn** [reply]
After following above methods, I'm getting below errors when running "ng build --watch=true" in client folder

<img width="785" alt="image" src="https://github.com/Sunbird-Ed/Community/assets/125034682/8ef7e856-f8be-492d-b152-f0555cedcde9">

**Reply by rajeshkumaravel** [reply]
@israr-trigyn 
Can you please share the node, npm and angular version (Global) .
CC @vinukumar-vs @rajeevsathish @anandp504

**Reply by israr-trigyn** [reply]
@rajeshkumaravel 

**NODE:-** v16.16.0
**NPM :-** 8.11.0
**Angular:-** 15.2.8

I tried using Node version "14.20.1" but still i was getting above errors.

**Reply by israr-trigyn** [reply]
@rajeshkumaravel 

Please response to above query.

I once tried downgrading the node version to "14.20.1" but was facing the same errors as shown above while running build generation command.

and after i switched to Node version "16.16.0" build command working fine but the only problem is hot reload not working.

CC @vinukumar-vs  @rajeshkumaravel  @anandp504

**Comment by abhishekpnt** [comment]
@israr-trigyn Please try to do the following

- a fresh clone of the portal with Angular-11(Global),Node 14.20
- Downgrade the moment version to 2.24.0 in package json of client (Both in dependencies and resolutions)
- Run yarn install

CC @vinukumar-vs @rajeevsathish

**Reply by israr-trigyn** [reply]
<img width="813" alt="image" src="https://github.com/Sunbird-Ed/Community/assets/125034682/7009f1d7-e728-444a-bed1-cb2e1a0bead7">



Getting above errors while running "yarn install" in client folder when using below versions:

node: 14.20.0
angular: 11

**Reply by abhishekpnt** [reply]
@israr-trigyn This error message clearly shows that the error is due to node-gyp library. Please try to configure that on your machine.

Since you are doing the installation on a windows machine, please refer this for the setup
https://github.com/orgs/Sunbird-Ed/discussions/433

CC : @rajeevsathish @vinukumar-vs

**Reply by israr-trigyn** [reply]
@abhishekpnt 

I followed methods:

1. install node-gyp globally
2. install node version 14.20.1
3. installed angular version 11
4. python version 3

After following above methods I'm successfully able to run **yarn install** but still when running **ng build --watch=true** hot reload not working.

**NOTE:** Running **ng build --watch=true** doesn't show the changes but if i run **ng build --watch=true --configuration=staging** then it does show the changes. but hot reload still doesn't work.

**Reply by vinukumar-vs** [reply]
@israr-trigyn as we discussed over the call, it is specific to individuals. Another member from your team able to run it properly.

I am considering this issue is resolved internally after you sync up with your team member.

**Reply by israr-trigyn** [reply]
@vinukumar-vs @rajeshkumaravel @abhishekpnt 

We're also following the same methods one of our team mate has followed and using the same versions too. still hot reload not working,

After changing files getting error warning in client folder:

<img width="698" alt="image" src="https://github.com/Sunbird-Ed/Community/assets/125034682/b02e6afa-eb83-4790-9b20-af7632d12e5b">

**Reply by israr-trigyn** [reply, marked as answer]
Issue resolved thanks.
